nautobot/docs/configuration/authentication/sso.md

```diff
@@ -130,12 +175,142 @@ information please utilize these additional resources.
+6. Skip step 10 in the instructions, as a signed response is not required.
+7. For step 11 of the instructions, under *Name ID*, set the following:
+    - **Name ID Format**: Select *EMAIL*
+    - **Name ID:** Select *Basic Information > Primary Email*
+8. For step 13 of the instructions, on the *Attribute mapping* page, add the following mappings for *Google Directory attributes* to *App attributes*:
+    - *Primary email* --> `email`
+    - *First name* --> `first_name`
+    - *Last name* --> `last_name`
+9. Click *Finish*
+
+#### Configure Nautobot
+
+There is a lot to configure to inform Nautobot how to integrate with SAML, so please provide the following configuration very carefully. All of these values must be correct in your `nautobot_config.py`.
+
+!!! important
+	Refer to the [official Python Social Auth documentation for required SAML configuration](https://python-social-auth.readthedocs.io/en/latest/backends/saml.html#required-configuration) if you run into any issues.
+
+```python
+# Django authentication backends
+AUTHENTICATION_BACKENDS = [
+    "social_core.backends.saml.SAMLAuth",
+    "nautobot.core.authentication.ObjectPermissionBackend",
+]
+
+# The https FQDN to your Nautobot instance
+SOCIAL_AUTH_SAML_SP_ENTITY_ID = "https://nautobot.example.com/"
+
+# X.509 cert/key pair used for host verification are not used for this example because
+# Nautobot is directly authenticating itself to Google. Set them to empty strings.
+SOCIAL_AUTH_SAML_SP_PUBLIC_CERT = ""
+SOCIAL_AUTH_SAML_SP_PRIVATE_KEY = ""
+
+# A dictionary that contains information about your app. You must specify values for
+# English at a minimum.
+SOCIAL_AUTH_SAML_ORG_INFO = {
+    "en-US": {
+        "name": "Nautobot",
+        "displayname": "Nautobot",
+        "url": "https://nautobot.example.com",
+    }
+}
+
+# Technical point of contact
+SOCIAL_AUTH_SAML_TECHNICAL_CONTACT = {
+    "givenName": "Bob Jones",
+    "emailAddress": "bob@example.com"
+}
+
+# Support point of contact
+SOCIAL_AUTH_SAML_SUPPORT_CONTACT = {
+    "givenName": "Alice Jenkins",
+    "emailAddress": "alice@example.com"
+}
+
+# The Entity ID URL for Google from step 3
+GOOGLE_ENTITY_ID = "<Entity ID from Google>"
+
+# The SSO URL for Google from step 3
+GOOGLE_SSO_URL = "<SSO URL from Google>"
+
+# The Certificate for Google from step 3
+GOOGLE_CERTIFICATE = "<Certificate from Google>"
+
+# The most important setting. List the Entity ID, SSO URL, and x.509 public key certificate
+# for each provider that you app wants to support. We are only supporting Google for this
+# example.
+SOCIAL_AUTH_SAML_ENABLED_IDPS = {
+    "google": {
+        "entity_id": GOOGLE_ENTITY_ID,
+        "url": GOOGLE_SSO_URL,
+        "x509cert": GOOGLE_CERTIFICATE,
+        # These are used to map Google attribute fields to User object fields in Nautobot
+        "attr_user_permanent_id": "email",
+        "attr_first_name": "first_name",
+        "attr_last_name": "last_name",
+        "attr_username": "email",
```

Comment:
Not `"username"`?

Reply:
The `attr_*` keys correspond to the keys specified as attributes in Google (or the IDP) and now that you point this out, I don' think this key even does anything for Google.